### PR TITLE
fix: various bugs in ledger integration

### PR DESCRIPTION
**Pull request by zoltana-ledger**
**Description**
Addressed below issues. Additionally Swaps and other action modals are not working with the confirmation UI.

**Issue**
- Better error messaging around supported operations
https://github.com/MetaMask/metamask-mobile/issues/4454
- Device now says busy if there is a pending transactions
https://github.com/MetaMask/metamask-mobile/issues/4457
- Rejection of signing a transaction does not cause disconnected state anymore
https://github.com/MetaMask/metamask-mobile/issues/4458
- Everything uses ReusableModals
https://github.com/MetaMask/metamask-mobile/issues/4466
- Try button now works
https://github.com/MetaMask/metamask-mobile/issues/4467
- Partially addressed some UI issues:
https://github.com/MetaMask/metamask-mobile/issues/4594
- Better error messaging around edge cases of pairing issues
https://github.com/MetaMask/metamask-mobile/issues/4596

**Checklist**

* [x] There is a related GitHub issue
* [x] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: app/components/Views/ApproveView/Approve/index.js [hunk 1]
```diff
@@ -518,6 +533,18 @@ class Approve extends PureComponent {
       const finalizeConfirmation = (confirmed) => {
         if (!confirmed) {
           this.setState({ transactionHandled: false });
+
+          // If a rejection happened on the ledger UI we need to cancel the transaction
+          // In any othercase component will unmount takes care of it
```
Comment (gantunesr) [inline app/components/Views/ApproveView/Approve/index.js:538]: nit

```suggestion
          // In any other case, componentWillUnmount will take care of it
```

File: app/components/Views/Approval/index.js [hunk 2]
```diff
@@ -128,13 +128,22 @@ class Approval extends PureComponent {
...
+          // We hand over execution to the ledger flow it'll take care of cancelling
+          if (!isLedgerAccount) {
+            Engine.context.TransactionController.cancelTransaction(
+              transaction.id,
+            );
+
+            Engine.context.TransactionController.hub.removeAllListeners(
+              `${transaction.id}:finished`,
+            );
+          }
```
Comment (gantunesr) [inline app/components/Views/Approval/index.js:144]: This piece of code is repeated multiple times, it could be a method

**Comment by GummyGod** [comment]
I have read the CLA Document and I hereby sign the CLA
